components/brave_wallet/browser/hd_keyring.cc

```diff
@@ -157,6 +159,59 @@ std::vector<uint8_t> HDKeyring::SignMessage(const std::string& address,
   return signature;
 }
 
+bool HDKeyring::RecoverAddress(const std::vector<uint8_t>& message,
+                               const std::vector<uint8_t>& signature,
+                               std::string* address) {
+  CHECK(address);
+  // A compact ECDSA signature (recovery id byte + 64 bytes).
+  if (signature.size() != 65)
+    return false;
+
+  std::vector<uint8_t> signature_only = signature;
+  uint8_t v = signature_only.back();
+  if (v < 27) {
+    VLOG(1) << "v should be >= 27";
+    return false;
+  }
+
+  // v = chain_id ? recid + chain_id * 2 + 35 : recid + 27;
+  // So recid = v - 27 when chain_id is 0
+  uint8_t recid = v - 27;
+  signature_only.pop_back();
+
+  // Get the 32 byte message hash
+  std::vector<uint8_t> hash;
+  std::string prefix("\x19");
+  prefix += std::string("Ethereum Signed Message:\n" +
+                        base::NumberToString(message.size()));
+  std::vector<uint8_t> hash_input(prefix.begin(), prefix.end());
+  hash_input.insert(hash_input.end(), message.begin(), message.end());
+  hash = KeccakHash(hash_input);
```

Comment:
We should put this into an internal function to be shared with `HDKeyring::SignMessage`

Reply:
Fixed in rebase https://github.com/brave/brave-core/pull/11590/commits/83d9229f9bd1c95362238c415c7c02f7f6f6c868
See `GetMessageHash`